ipam/provider/aws.go

```diff
@@ -0,0 +1,258 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"text/template"
+	"time"
+
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/e2e-harness/pkg/framework"
+	awsclient "github.com/giantswarm/e2eclients/aws"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// apiextensionsAWSConfigE2EChartValues is modified version of
+// e2etemplates.ApiextensionsAWSConfigE2EChartValues for IPAM test scenario.
+// Major difference is lack of subnet fields and need to replace clusterName
+// format string value. ClusterName is not based on env variable because there
+// would be too high risk for unintentional disruption of other tests if
+// ${CLUSTER_NAME} would be dynamically changed on the fly.
+const apiextensionsAWSConfigE2EChartValues = `commonDomain: {{ .CommonDomain }}
+clusterName: {{ .ClusterName }}
+clusterVersion: v_0_1_0
+sshPublicKey: {{ .SSHPublicKey }}
+versionBundleVersion: {{ .VersionBundleVersion }}
+aws:
+  region: {{ .AWSRegion }}
+  apiHostedZone: {{ .AWSAPIHostedZoneGuest }}
+  ingressHostedZone: {{ .AWSIngressHostedZoneGuest }}
+  routeTable0: {{ .AWSRouteTable0 }}
+  routeTable1: {{ .AWSRouteTable1 }}
+  vpcPeerId: {{ .AWSVPCPeerID }}
+`
```

Comment:
Don't we have these things in the `apiextensions` repo? I would like to try defining these things in one place and not over and over again here and there.

Reply:
There's actually comment describing that this originates from `e2etemplates`, but this test requires changes so hence copying. I'll see if I could somehow incorporate these to source repo so it would be only in one place.